pkg/runner/retriever/retriever.go

```diff
@@ -35,15 +35,17 @@ var (
 )
 
 // Retriever is a function that retrieves the MCP server definition from the registry.
-type Retriever func(context.Context, string, string, string) (string, registry.ServerMetadata, error)
+type Retriever func(context.Context, string, string, string, string) (string, registry.ServerMetadata, error)
 
 // GetMCPServer retrieves the MCP server definition from the registry.
-func GetMCPServer(
+func GetMCPServer( //nolint:gocyclo
```

Comment:
It looks like most of the code for group logic is in a big `if` statement. It might be simpler to introduce a new `GetMCPServerFromGroupCatalog` and have some logic duplicated instead of a big `if` statement. Also the logic for `IsImageProtocolScheme` will never apply to groups, so the duplication will be quite small and all of it will be calling helpers.